Cabal/tests/ParserTests.hs

```diff
@@ -88,6 +104,28 @@ regressionTest fp = cabalGoldenTest fp correct $ do
     input = "tests" </> "ParserTests" </> "regressions" </> fp
     correct = replaceExtension input "format"
 
+formatRoundTripTest :: FilePath -> TestTree
+formatRoundTripTest fp = testCase "roundtrip" $ do
+    contents <- BS.readFile input
+    x <- parse contents
+    let contents' = showGenericPackageDescription x
+    y <- parse (toUTF8BS contents')
+    let y' = if x ^. L.packageDescription . L.license == UnspecifiedLicense
```

Comment:
It will hopefully go away in SPDX change. The short explanation is that `parse . pretty` roundtrip on `License` doesn't hold ATM. Yet Hackage never accepts a package with `UnspecifiedLicense`, so I didn't bother to refactor it yet.